ciao-controller/internal/datastore/datastore.go

```diff
@@ -845,6 +845,12 @@ func (ds *Datastore) GetAllInstancesByNode(nodeID string) ([]*types.Instance, er
 // AddInstance will store a new instance in the datastore.
 // The instance will be updated both in the cache and in the database
 func (ds *Datastore) AddInstance(instance *types.Instance) error {
+	err := ds.db.addInstance(instance)
+
+	if err != nil {
+		return errors.Wrap(err, "Error adding instance to database")
+	}
+
 	// add to cache
 	ds.instancesLock.Lock()
```

Comment:
do you think we should lock the cache, update the persistent store, update the cache, unlock the cache? Can't decide if this is paranoid overkill.